src/main/java/com/mcmoddev/mmdbot/modules/commands/info/CmdTranslateMappings.kt

```diff
@@ -0,0 +1,205 @@
+package com.mcmoddev.mmdbot.modules.commands.info
+
+import com.jagrosh.jdautilities.command.Command
+import com.jagrosh.jdautilities.command.CommandEvent
+import com.mcmoddev.mmdbot.core.Utils
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import me.shedaniel.linkie.*
+import me.shedaniel.linkie.namespaces.MCPNamespace
+import me.shedaniel.linkie.namespaces.MojangNamespace
+import me.shedaniel.linkie.namespaces.YarnNamespace
+import me.shedaniel.linkie.utils.MappingsQuery
+import me.shedaniel.linkie.utils.QueryContext
+import me.shedaniel.linkie.utils.ResultHolder
+import me.shedaniel.linkie.utils.tryToVersion
+import net.dv8tion.jda.api.EmbedBuilder
+import net.dv8tion.jda.api.entities.MessageEmbed
+import net.dv8tion.jda.api.events.interaction.ButtonClickEvent
+import net.dv8tion.jda.api.hooks.ListenerAdapter
+import net.dv8tion.jda.api.interactions.components.Button
+import java.awt.Color
+import java.util.*
+
+class CmdTranslateMappings(name: String, private val namespace1: Namespace, private val namespace2: Namespace, vararg aliases: String?) : Command() {
+    init {
+        this.name = name.lowercase(Locale.ROOT)
+        this.aliases = aliases
+        help = "Turn $namespace1 into $namespace2"
+    }
+
+    /**
+     * @param event The [CommandEvent] that triggered this Command.
+     */
+    override fun execute(event: CommandEvent) {
+        if (!Utils.checkCommand(this, event)) return
+        if (event.args.isEmpty()) {
+            event.channel.sendMessage("No arguments given!").queue()
+            return
+        }
+
+        val args = event.args.split(' ');
+
+        val query = args[0];
+        val version = args.getOrElse(1) {
+            val namespace2Versions = namespace2.getAllVersions().toSet()
+            namespace1.getAllVersions().filter { version -> namespace2Versions.contains(version) }.maxWithOrNull(nullsFirst(compareBy { it.tryToVersion() }))!!
+        }
+
+        scope.launch {
+            val originProvider = namespace1.getProvider(version)
+            val targetMappings = namespace2.getProvider(version).get()
+            var hasPerfectMatch = false
+            var embeds = (MappingsQuery.queryClasses(
+                QueryContext(
+                    originProvider,
+                    query
+                )
+            ).value.asSequence() + MappingsQuery.queryMember(
+                QueryContext(originProvider, query)
+            ) { it.members.asSequence() }.value.asSequence())
```

Comment:
There's a bit of inconsistent styling here

Reply:
Looks a little nicer now